### ISSUE TITLE
helm package: No error printed on invalid semantic version

### ISSUE THREAD

**Issue by ammario** [helm/helm#8864]
This leads to confusing errors when we aren't setting versions correctly. E.g:

```
$ helm search repo --debug -l coder-nightly
No results found
```

See 

https://github.com/helm/helm/blob/fc9b46067f8f24a90b52eba31e09b31e69011e93/pkg/repo/index.go#L187-L196

**Comment by hickeyma** [comment]
Thanks for raising the issue @ammario. Can you describe in more detail what you mean by "confusing errors"? What do you expect to happen?

I am asking to get a better understanding of your issue as search command works as expected for me.

**Comment by mattfarina** [comment]
@ammario Do you use `helm lint` on your chart before packaging it?

**Comment by bacongobbler** [comment]
> Do you use `helm lint` on your chart before packaging it?

Interesting... It appears that it is OK to package a chart with an invalid semver. That does not appear to be expected behaviour.

```
/tmp ><> helm create test-8864
Creating test-8864
/tmp ><> cd test-8864/
/tmp/test-8864 ><> vim Chart.yaml
/tmp/test-8864 ><> cat Chart.yaml
apiVersion: v2
name: test-8864
description: A Helm chart for Kubernetes
type: application
version: foo
appVersion: 1.16.0
/tmp/test-8864 ><> helm package .
Successfully packaged chart and saved it to: /tmp/test-8864/test-8864-foo.tgz
```

`helm lint` catches it though.

```
/tmp/test-8864 ><> helm lint .
==> Linting .
[ERROR] Chart.yaml: version 'foo' is not a valid SemVer
[INFO] Chart.yaml: icon is recommended

Error: 1 chart(s) linted, 1 chart(s) failed
```

I thought that was one of the scenarios we were supposed to catch on `helm package`, as the rest of Helm relies on the package being a valid semver to perform things like automatic upgrades to the latest version by default.

We cannot capture and return error output on `helm search`. It is intentional for Helm to silently ignore packages that are not valid semver. If one package in the repository index is failing a semver check, the entire chart userbase cannot pull any charts from that repository.

**Comment by bacongobbler** [comment]
So.... Looking at this further, I'd propose we do not look at addressing the `helm search` case, but instead look at the ways where a user can package a Helm chart with an invalid semver. That should prevent users from packaging and uploading invalid charts in the first place.

**Comment by ammario** [comment]
@bacongobbler I understand and agree with that DoS angle. I think your proposal of enforcing SemVer in the helm package step is a great solution.

**Comment by bacongobbler** [comment]
Fixed in #8933.
